src/silvimetric/resources/metric.py

```diff
@@ -39,7 +40,8 @@ def to_json(self) -> dict[str, any]:
             'name': self.name,
             'dtype': np.dtype(self.dtype).str,
             'dependencies': self.dependencies,
-            'method': getsource(self._method)
+            'method_str': getsource(self._method),
+            'method': base64.b64encode(pickle.dumps(self._method)).decode()
```

Comment:
Too much stuff happening in this one line.

I think you should maybe check if you can pickle the method before you do everything else. It's very possible that the user provided a method that doesn't pickle or is invalid. Some kind of checking on that needs to happen here before we insert it into the db.